src/structures/Chat.js

```diff
@@ -190,19 +192,33 @@ class Chat extends Base {
                 if (searchOptions && searchOptions.fromMe && m.id.fromMe !== searchOptions.fromMe) {
                     return false;
                 }
+
+                if (searchOptions && searchOptions.minTimestamp && m.t <= searchOptions.minTimestamp) {
+                    return false;
+                }
+
+                if (searchOptions && searchOptions.maxTimestamp && m.t >= searchOptions.maxTimestamp) {
+                    return false;
+                }
+
                 return true;
             };
 
+
             const chat = window.Store.Chat.get(chatId);
             let msgs = chat.msgs.getModelsArray().filter(msgFilter);
 
             if (searchOptions && searchOptions.limit > 0) {
                 while (msgs.length < searchOptions.limit) {
                     const loadedMessages = await window.Store.ConversationMsgs.loadEarlierMsgs(chat);
                     if (!loadedMessages || !loadedMessages.length) break;
+
+                    if (searchOptions.minTimestamp && loadedMessages.some(msg => msg.t < searchOptions.minTimestamp))
+                        break;
+
                     msgs = [...loadedMessages.filter(msgFilter), ...msgs];
                 }
-                
+
                 if (msgs.length > searchOptions.limit) {
```

Comment:
Extraneous chats with timestamp beyond range are handled with splice  ?

Reply:
why would this be a problem? can you elaborate?

Reply:
So the loop breaks before we insert the loadedMessages that was filtered, some messages might be within the range but not returned, maybe line 219 should go first . Unless I missed something

Reply:
You are right. I will fix it.

Reply:
@LucianoHanna where we standing with this?